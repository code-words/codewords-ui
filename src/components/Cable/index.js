import React, { Fragment } from 'react';
import { ActionCable } from 'react-actioncable-provider';
const shortid = require('shortid');


const Cable = ({ conversations, handleReceivedMessage }) => {
  return (
    <Fragment>
      {conversations.map(conversation => {
        return (
          <ActionCable
          id={conversation.id}
            key={shortid.generate()}  
            channel={{ channel: 'MessagesChannel', conversation: conversation.id }}
            onReceived={handleReceivedMessage}
          />
        );
      })}
    </Fragment>
  );
};

export default Cable;