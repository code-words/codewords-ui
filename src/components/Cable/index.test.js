import Cable from './index';
import React from 'react';
import { shallow } from 'enzyme';


describe('Cable', () => {
    let mockData = ["test", "another test"]
    let wrapper = shallow(<Cable conversations= {mockData}/>)
    it('should match Snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });
});
