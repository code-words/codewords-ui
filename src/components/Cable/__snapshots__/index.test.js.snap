// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cable should match Snapshot 1`] = `ShallowWrapper {}`;
